### DISCUSSION TITLE
May I ask how to make HID devices and Serial devices coexist?

### DISCUSSION THREAD

**Discussion by Mars-CN** [hathach/tinyusb#2319] [General]
I am using TinyUSB for an application, and the requirement is to provide an HID keyboard and a Serial device to the PC through USB.

The IC I am using is ESP32 (although it doesn't matter). According to the example, starting the serial port and keyboard separately is not a problem. However, after I create a configuration descriptor for the HID keyboard and serial port, these two devices cannot be recognized on the PC. Can HID devices coexist with Serial devices? Is there a problem with my design?

How should I achieve it?

**Comment by hathach** [comment]
yes, just create a correct configuration descriptor for both interfaces.
